services/consensuscontext/create_transactions_block.go

```diff
@@ -62,10 +75,42 @@ func (s *service) createTransactionsBlock(ctx context.Context, input *services.R
 	return txBlock, nil
 }
 
-func (s *service) createTriggerTransaction(blockTime primitives.TimestampNano) *protocol.SignedTransaction {
+func (s *service) proposeBlockReferenceTime(ctx context.Context, prevReferenceTime primitives.TimestampSeconds) (primitives.TimestampSeconds, error) {
```

Comment:
missing monotonicity requirement: proposeBlockReferenceTime = max(prevReferenceTime, management. GetCurrentReference)

Reply:
as leader i create proposed time from getCurrent, then in line 83 i go to the validation , where i check that propose >= prev. 

please look at code again.

Reply:
done